backend/src/lastwill/utilities/lastwill-templating.util.spec.ts

```diff
@@ -0,0 +1,451 @@
+      const res = generateLocationHeader(null)
+      // ASSERT
+      expect(res).toContain(PLACEHOLDERS.CITY)
+    })
+  })
+
+  describe('generateInitialText', () => {
+    it('should generate header', () => {
+      // ACT
+      const res = generateInitialText('testName', '01.01.1970', 'coruscant')
+      // ASSERT
+      expect(res).toContain('testName')
+      expect(res).toContain('01.01.1970')
+      expect(res).toContain('coruscant')
+    })
+
+    it('should use placeholders', () => {
+      // ACT
+      const res = generateInitialText(null, null, null)
+      // ASSERT
+      expect(res).toContain(PLACEHOLDERS.PERSON_NAME)
+      expect(res).toContain(PLACEHOLDERS.BIRTH_DATE)
+      expect(res).toContain(PLACEHOLDERS.BIRTH_PLACE)
+    })
+  })
+
+  describe('generateInheritanceForOrganisation', () => {
+    it('should create valid paragraph', () => {
+      // ACT
+      const res = generateInheritanceForOrganisation(SAMPLE_ORGANISATION)
+      // ASSERT
+      expect(res).toContain(SAMPLE_ORGANISATION.name)
+      expect(res).toContain(SAMPLE_ORGANISATION.percentage.toString())
+      expect(res).toContain(SAMPLE_ORGANISATION.address.city)
+      expect(res).toContain(SAMPLE_ORGANISATION.address.zipCode)
+    })
+
+    it('should use placeholders for null values', () => {
+      // ACT
+      const res = generateInheritanceForOrganisation({} as Organisation)
+      // ASSERT
+      expect(res).toContain(PLACEHOLDERS.COMPANY_NAME)
+      expect(res).toContain(PLACEHOLDERS.PERCENTAGE)
+      expect(res).toContain(PLACEHOLDERS.CITY)
+      expect(res).toContain(PLACEHOLDERS.ZIP_CODE)
+    })
+  })
+
+  describe('generateInheritanceForPerson', () => {
+    it('should create valid paragraph', () => {
+      // ACT
+      const res = generateInheritanceForPerson(SAMPLE_PERSON)
+      // ASSERT
+      expect(res).toContain(SAMPLE_PERSON.name)
+      expect(res).toContain(SAMPLE_PERSON.birthDate)
+      expect(res).toContain(SAMPLE_PERSON.percentage.toString())
+    })
+
+    it('should use placeholder for null values', () => {
+      // ACT
+      const res = generateInheritanceForPerson({} as Person)
+      // ASSERT
+      expect(res).toContain(PLACEHOLDERS.PERSON_NAME)
+      expect(res).toContain(PLACEHOLDERS.BIRTH_DATE)
+      expect(res).toContain(PLACEHOLDERS.PERCENTAGE)
+    })
+  })
+
+  describe('generateFinancialInheritancePragraphs', () => {
+    it('should return valid paragraph for person', () => {
+      // ACT
+      // Use first as the following is legal giberish
+      const res = generateFinancialInheritancePragraphs(
+        [SAMPLE_PERSON],
+        [SAMPLE_FINANCIAL_ASSET],
+      )[0]
+      // ASSERT
+      expect(res.title).toEqual(PARAGRAPH_TITLES.FINANCIAL)
+      expect(
+        res.contents.find(
+          (e) =>
+            e.includes(SAMPLE_PERSON.name) &&
+            e.includes(SAMPLE_PERSON.percentage.toString()) &&
+            e.includes(SAMPLE_PERSON.birthDate),
+        ),
+      ).toBeTruthy()
```

Comment:
```suggestion
      ).not.toBeUndefined()
```
array.find returns either the value or undefined, so the more logically test would be checking if it is not undefined, truthy works too but makes less sense